hello-robot/stretch_web_teleop
# WIP: Homing Button

# Description

\[TODO: describe, in-detail, what issue this PR addresses and how it addresses it. Link to relevant Github Issues.\]

# Testing procedure

 1. Start with an un-homed robot. You can power cycle the robot, or a quick way is to call `stretch_robot_dynamixel_reboot.py`.
 2. In your Ament workspace, go to Stretch ROS2 and switch to the `bugfix/priority_mode_stall` branch. If PR https://github.com/hello-robot/stretch_ros2/pull/162 is merged, you can `git pull` on the default branch.
 3. In your Ament workspace, go to Stretch Web Teleop and switch to the `feature/homing` branch.
 4. In the same repo, cherry-pick a numpy fix onto the homing branch. The command is `git cherry-pick -n 6dfcf32e50175dff5cfabaa2c581486660f1d143`, followed by `git reset`. If PR https://github.com/hello-robot/stretch_web_teleop/pull/97 is merged, then this step may be unnecessary (depends on whether I've rebased this branch).
 5. `colcon build` the Ament workspace as usual
 6. Run the interface as usual

The homing button will appear. Click it and it should disappear after homing is complete.

# Before opening a pull request

From the top-level of this repository, run:


# To merge
